examples/rust/opensk/src/lib.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! TODO: Add docs
+
+#![no_std]
+wasefire::applet!();
+
+use opensk_lib::api::attestation_store::{Attestation, AttestationStore, Id};
+use opensk_lib::api::connection::HidConnection;
+use opensk_lib::api::crypto::software_crypto::SoftwareCrypto;
+// use opensk_lib::api::customization::{AAGUID_LENGTH, DEFAULT_CUSTOMIZATION};
+use opensk_lib::api::customization::CustomizationImpl;
+use opensk_lib::env::Env;
+use persistent_store::Store;
+// use wasefire::clock::{Handler, Timer};
+
+mod clock;
+mod keystore;
+mod rng;
+mod storage;
+mod user_presence;
+mod write;
+
+fn main() {
+    debug!("hello world");
+}
+
+// pub const AAGUID: &[u8; AAGUID_LENGTH] = todo!();
+// include_bytes!(concat!(env!("OUT_DIR"), "/opensk_aaguid.bin"));
+
+// const WASEFIRE_CUSTOMIZATION: CustomizationImpl =
+//     CustomizationImpl { aaguid: AAGUID, ..DEFAULT_CUSTOMIZATION };
+
+#[derive(Clone, Copy)]
+pub struct WasefireHidConnection;
+
+impl HidConnection for WasefireHidConnection {
+    fn send_and_maybe_recv(
+        &mut self, buf: &mut [u8; 64], timeout_ms: usize,
+    ) -> opensk_lib::api::connection::SendOrRecvResult {
+        todo!()
+    }
+}
+
+impl Default for WasefireHidConnection {
+    fn default() -> Self {
+        Self {}
+    }
+}
+
+struct WasefireEnv {
+    write: write::WasefireWrite,
+    // store: Store<Storage<S, C>>,
+    storage: storage::WasefireStorage,
+    // upgrade_storage: Option<UpgradeStorage<S, C>>,
+    main_connection: WasefireHidConnection,
+    vendor_connection: WasefireHidConnection,
+    // blink_pattern: usize,
+    clock: clock::WasefireClock,
+    // c: PhantomData<C>,
+}
+
+impl AttestationStore for WasefireEnv {
+    fn get(
+        &mut self, id: &Id,
+    ) -> Result<Option<Attestation>, opensk_lib::api::attestation_store::Error> {
+        todo!()
+    }
+
+    fn set(
+        &mut self, id: &Id, attestation: Option<&Attestation>,
+    ) -> Result<(), opensk_lib::api::attestation_store::Error> {
+        todo!()
+    }
+}
+
+impl Default for WasefireEnv {
```

Comment:
You can derive `Default`. Same in other files. That would remove some additional boilerplate.

```rust
#[derive(Default)]
struct WasefireEnv { ... }
```

Reply:
Done.